code/extension/Member.php

```diff
@@ -26,6 +26,8 @@ class Member extends \DataExtension
 
     private static $admins_can_disable = false;
 
+    private static $regenerate_on_activation = true;
```

Comment:
Manual regeneration would indeed need to be implemented, but I think that could be left to the implementation, as it's probably an action on a front-end form with its own checking etc. in place.